x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/physical/local/InsertFieldExtraction.java

```diff
@@ -58,9 +65,24 @@ public PhysicalPlan apply(PhysicalPlan plan) {
 
             // add extractor
             if (missing.isEmpty() == false) {
-                // collect source attributes and add the extractor
-                var extractor = new FieldExtractExec(p.source(), p.child(), List.copyOf(missing));
-                p = p.replaceChild(extractor);
+                // identify child (for binary nodes) that exports doc values and place the field extractor there
```

Comment:
I find it confusing to name this "doc values". Isn't this about doc id? (`EsQueryExec.DOC_ID_FIELD`)